indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -575,58 +542,231 @@ public Response getCompleteTasks(
       @Context final HttpServletRequest req
   )
   {
-    final Function<String, Task> taskFunction = taskId -> {
-      final Optional<Task> optionalTask = taskStorageQueryAdapter.getTask(taskId);
-      if (!optionalTask.isPresent()) {
+    return getTasks("complete", null, null, maxTaskStatuses, null, req);
+  }
+
+  @GET
+  @Path("/tasks")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getTasks(
+      @QueryParam("state") final String state,
+      @QueryParam("datasource") final String dataSource,
+      @PathParam("interval") final String interval,
+      @QueryParam("max") final Integer maxCompletedTasks,
+      @QueryParam("type") final String type,
+      @Context final HttpServletRequest req
+  )
+  {
+
+    //check for valid state
+    if (state != null) {
+      List validStates = ImmutableList.of("pending", "waiting", "running", "complete");
+      if (!validStates.contains(StringUtils.toLowerCase(state))) {
+        return Response.status(Status.BAD_REQUEST)
+                       .entity("Bad state query param passed : " + state).entity(ImmutableList.of()).build();
```

Comment:
I don't think you can do `entity` twice. I think the last one wins if you do this, so you should get rid of the second `entity`. And while doing that, you might as well make the message a little nicer, something like:

```java
String.format("Invalid 'state', valid values are: " + API_TASK_STATES)
```

Reply:
yeah, missed that there was `entity` twice here.